In [10]:
from math import *

In [11]:
#help(math)

In [13]:
e

2.718281828459045

In [14]:
inf

inf

In [ ]:

isfinite(x, /) : Return True if x is neither an infinity nor a NaN, and False otherwise.

isinf(x, /) : Return True if x is a positive or negative infinity, and False otherwise.

isnan(x, /) : Return True if x is a NaN (not a number), and False otherwise.


# Basic

In [ ]:
#floor(x, /):Return the floor of x as an Integral.
#This is the largest integer <= x.
    

In [ ]:
#fmod(x, y, /):Return fmod(x, y), according to platform C.
#x % y may differ.

In [ ]:
#exp(x, /):Return e raised to the power of x.

In [ ]:
#sqrt(x, /): Return the square root of x.
    

In [ ]:
#modf(x, /) : Return the fractional and integer parts of x.
        
        Both results carry the sign of x and are floats.

In [15]:
#factorial(x, /):Find x!.
factorial(5)

120

NameError: name 'comb' is not defined

# Trignomatric

π rad = 180°

1Rad × 180/π = 57.296Deg

α(degrees) = α(radians) × 180° / π

degrees = radians × 180° / π

<table class="ntable">
<tbody><tr>
<th>Radians (rad)</th>
<th>Radians (rad)</th>
<th>Degrees (°)</th>
</tr>
<tr>
<td>0 rad</td>
<td>0 rad</td>
<td>0°</td>
</tr>
<tr>
<td>π/6 rad</td>
<td>0.5235987756 rad</td>
<td>30°</td>
</tr>
<tr>
<td>π/4 rad</td>
<td>0.7853981634 rad</td>
<td>45°</td>
</tr>
<tr>
<td>π/3 rad</td>
<td>1.0471975512 rad</td>
<td>60°</td>
</tr>
<tr>
<td>π/2 rad</td>
<td>1.5707963268 rad</td>
<td>90°</td>
</tr>
<tr>
<td>2π/3 rad</td>
<td>2.0943951024 rad</td>
<td>120°</td>
</tr>
<tr>
<td>3π/4 rad</td>
<td>2.3561944902 rad</td>
<td>135°</td>
</tr>
<tr>
<td>5π/6 rad</td>
<td>2.6179938780 rad</td>
<td>150°</td>
</tr>
<tr>
<td>π rad</td>
<td>3.1415926536 rad</td>
<td>180°</td>
</tr>
<tr>
<td>3π/2 rad</td>
<td>4.7123889804 rad</td>
<td>270°</td>
</tr>
<tr>
<td>2π rad</td>
<td>6.2831853072 rad</td>
<td>360°</td>
</tr>
</tbody></table>

In [20]:
#degrees(x, /):Convert angle x from radians to degrees.
degrees(1.5707963267948966)

90.0

In [19]:
#radians(x, /):Convert angle x from degrees to radians.
radians(90)

1.5707963267948966

## rad

In [23]:
#sin(x, /) : Return the sine of x (measured in radians).   
sin(1.5707963267948966)

1.0

In [25]:
sin(0)

0.0

In [24]:
#cos(x, /) : Return the cosine of x (measured in radians).
cos(0)

1.0

In [26]:
cos(1)

0.5403023058681398

In [ ]:
#tan(x, /) : Return the tangent of x (measured in radians).

In [ ]:
#asin(x, /) : Return the arc sine (measured in radians) of x.  

In [ ]:
#acos(x, /) : Return the arc cosine (measured in radians) of x.    

In [ ]:
#atan(x, /) : Return the arc tangent (measured in radians) of x.

In [ ]:
#atan2(y, x, /) : Return the arc tangent (measured in radians) of y/x.
        
        Unlike atan(y/x), the signs of both x and y are considered.
    

##  hyperbolic

In [ ]:
#sinh(x, /) : Return the hyperbolic sine of x.

In [ ]:
#cosh(x, /) : Return the hyperbolic cosine of x.

In [ ]:
#tanh(x, /):Return the hyperbolic tangent of x.

In [ ]:
#asinh(x, /) : Return the inverse hyperbolic sine of x.

In [ ]:
#acosh(x, /) : Return the inverse hyperbolic cosine of x.

In [ ]:
#atanh(x, /) : Return the inverse hyperbolic tangent of x.
    

# log(...)

In [ ]:
 : log(x, [base=math.e])
        Return the logarithm of x to the given base.
        
        If the base not specified, returns the natural logarithm (base e) of x.
    
    log10(x, /)
        Return the base 10 logarithm of x.
    
    log1p(x, /)
        Return the natural logarithm of 1+x (base e).
        
        The result is computed in a way which is accurate for x near zero.
    
    log2(x, /)
        Return the base 2 logarithm of x.
    

# Function

In [ ]:
#ceil(x, /):Return the ceiling of x as an Integral.
#his is the smallest integer >= x.
    

In [ ]:
copysign(x, y, /):Return a float with the magnitude (absolute value) of x but the sign of y.

    On platforms that support signed zeros, copysign(1.0, -0.0)
    returns -1.0.


In [ ]:
#lgamma(x, /):Natural logarithm of absolute value of Gamma function at x.

In [ ]:
#erf(x, /):Error function at x.
#erfc(x, /):Complementary error function at x.
#expm1(x, /):Return exp(x)-1.
This function avoids the loss of precision involved in the direct evaluation of exp(x)-1 for small x.
    

In [ ]:
#fabs(x, /):Return the absolute value of the float x.
   

In [ ]:
frexp(x, /):Return the mantissa and exponent of x, as pair (m, e).
        m is a float and e is an int, such that x = m * 2.**e.
        If x is 0, m and e are both 0.  Else 0.5 <= abs(m) < 1.0.
    
    fsum(seq, /):Return an accurate floating point sum of values in the iterable seq.
        
        Assumes IEEE-754 floating point arithmetic.
    
    gamma(x, /):Gamma function at x.
    
    gcd(x, y, /):greatest common divisor of x and y
    
    hypot(x, y, /)
        Return the Euclidean distance, sqrt(x*x + y*y).
    
    isclose(a, b, *, rel_tol=1e-09, abs_tol=0.0)
        Determine whether two floating point numbers are close in value.
        
          rel_tol
            maximum difference for being considered "close", relative to the
            magnitude of the input values
          abs_tol
            maximum difference for being considered "close", regardless of the
            magnitude of the input values
        
        Return True if a is close in value to b, and False otherwise.
        
        For the values to be considered close, the difference between them
        must be smaller than at least one of the tolerances.
        
        -inf, inf and NaN behave similarly to the IEEE 754 Standard.  That
        is, NaN is not close to anything, even itself.  inf and -inf are
        only close to themselves.

    ldexp(x, i, /)
        Return x * (2**i).
        
        This is essentially the inverse of frexp().
    
        pow(x, y, /)
        Return x**y (x to the power of y).
        

    

    

    

    
    remainder(x, y, /)
        Difference between x and the closest integer multiple of y.
        
        Return x - n*y where n*y is the closest integer multiple of y.
        In the case where x is exactly halfway between two multiples of
        y, the nearest even value of n is used. The result is always exact.



    
    trunc(x, /)
        Truncates the Real x to the nearest Integral toward 0.
        
        Uses the __trunc__ magic method.


FILE
    (built-in)

# own

In [ ]:
com